types/evidence_test.go

```diff
@@ -638,58 +660,55 @@ func TestEvidenceProto(t *testing.T) {
 	}
 
 	tests := []struct {
-		testName string
-		evidence Evidence
-		wantErr  bool
-		wantErr2 bool
+		testName     string
+		evidence     Evidence
+		toProtoErr   bool
+		fromProtoErr bool
 	}{
-		{"&DuplicateVoteEvidence empty fail", &DuplicateVoteEvidence{}, false, true},
-		{"&DuplicateVoteEvidence nil voteB", &DuplicateVoteEvidence{VoteA: v, VoteB: nil}, false, true},
-		{"&DuplicateVoteEvidence nil voteA", &DuplicateVoteEvidence{VoteA: nil, VoteB: v}, false, true},
-		{"&DuplicateVoteEvidence success", &DuplicateVoteEvidence{VoteA: v2, VoteB: v}, false, false},
-		{"&ConflictingHeadersEvidence empty fail", &ConflictingHeadersEvidence{}, false, true},
-		{"&ConflictingHeadersEvidence nil H2", &ConflictingHeadersEvidence{H1: h1, H2: nil}, false, true},
-		{"&ConflictingHeadersEvidence nil H1", &ConflictingHeadersEvidence{H1: nil, H2: h2}, false, true},
-		{"ConflictingHeadersEvidence empty fail", ConflictingHeadersEvidence{}, false, true},
-		{"ConflictingHeadersEvidence nil H2", ConflictingHeadersEvidence{H1: h1, H2: nil}, false, true},
-		{"ConflictingHeadersEvidence nil H1", ConflictingHeadersEvidence{H1: nil, H2: h2}, false, true},
-		{"ConflictingHeadersEvidence success", ConflictingHeadersEvidence{H1: h1, H2: h2}, false, false},
-		{"LunaticValidatorEvidence empty fail", LunaticValidatorEvidence{}, false, true},
-		{"LunaticValidatorEvidence only header fail", LunaticValidatorEvidence{Header: header1}, false, true},
-		{"LunaticValidatorEvidence only vote fail", LunaticValidatorEvidence{Vote: v}, false, true},
-		{"LunaticValidatorEvidence header & vote fail", LunaticValidatorEvidence{Header: header1, Vote: v}, false, true},
-		{"LunaticValidatorEvidence success", LunaticValidatorEvidence{Header: header1,
+		{"DuplicateVoteEvidence empty fail", &DuplicateVoteEvidence{}, false, true},
```

Comment:
could you add a nil case 
```suggestion
        {"nil fail", nil, true, true},
		{"DuplicateVoteEvidence empty fail", &DuplicateVoteEvidence{}, false, true},
```